### PR TITLE
Update to Modulemd 2.0

### LINKED ISSUE #63: Migrate to libmodulemd 2.0 API

### PR DESCRIPTION
**Pull request by r4f4**
Fixes #63

### CODE REVIEW

File: tests/pull-default-module/modules.yaml [hunk 1]
```diff
@@ -7,6 +7,11 @@ data:
   version: 20180904161631
   context: cafebabe
   arch: x86_64
+  summary: Just a test module
```
Comment (igor-raits) [inline tests/pull-default-module/modules.yaml:10]: how come that now summary/description/license are mandatory? modulemd spec says it is optional.

File: fus.c [hunk 2]
```diff
@@ -31,61 +31,82 @@ dep_or_rel (Pool *pool, Id dep, Id rel, Id op)
...
+      g_auto(GStrv) reqv = modulemd_dependencies_get_runtime_streams_as_strv (deps, *n);
+      Id r = parse_module_stream_requires (pool, *n, reqv);
+      require = dep_or_rel (pool, require, r, REL_AND);
+    }
+
+  return require;
+}
+
+static Id
+parse_module_build_requires (Pool                 *pool,
```
Comment (igor-raits) [inline fus.c]: as I can see, this is fully identical to function above except "buildtime" vs "runtime", can you just pass this function as an argument?

**Review by igor-raits — COMMENTED** [review]
apart from few comments inline, looks good. I'm not expert in mmdv2 API, @sgallagher can you review please?

File: tests/pull-default-module/modules.yaml (hunk 1 above)
Comment (sgallagher) [reply tests/pull-default-module/modules.yaml:10]: Nope, they've always been mandatory, see: https://github.com/fedora-modularity/libmodulemd/blob/master/spec.v2.yaml#L34

Comment (r4f4) [reply tests/pull-default-module/modules.yaml:10]: They were not generating an error when using v1 API but they do in v2, that's the difference.

File: fus.c (hunk 2 above)
Comment (r4f4) [reply fus.c]: Sure, I'll change it.

File: tests/pull-default-module/modules.yaml (hunk 1 above)
Comment (sgallagher) [reply tests/pull-default-module/modules.yaml:10]: That would have been a bug in v1. They should have been raising errors.

**Review by sgallagher — CHANGES_REQUESTED** [review]
Looks generally good to me, but please rebase it atop master. Right now, Github indicates that it has conflicts.

**Comment by r4f4** [comment]
Sure, it's because a PR has just been merged. Just give me a moment.

File: fus.c (hunk 2 above)
Comment (r4f4) [reply fus.c]: Done in the new update.

**Comment by r4f4** [comment]
Rebased on top of master and with @ignatenkobrain's requested change.

**Pull request merged**